packages/operator/pkg/trainer/trainer.go

```diff
@@ -87,51 +87,57 @@ func (mt *ModelTrainer) Setup() (err error) {
 
 	commitID := ""
 
+	workDir := mt.trainerConfig.OutputDir
+	if len(workDir) != 0 {
+		mt.log.Info("Change current working dir", "new worker dir", workDir)
+
+		if err := os.Chdir(workDir); err != nil {
+			mt.log.Error(err, "Changing current working dir failed",
+				"new worker dir", workDir,
+			)
+
+			return err
+		}
+	}
+
 	// Downloads a source code
-	switch k8sTraining.VCS.Spec.Type {
+	switch k8sTraining.AlgorithmSource.Conn.Spec.Type {
 	case connection.GITType:
 		commitID, err = mt.cloneUserRepo(k8sTraining, mt.trainerConfig.OutputDir)
 		if err != nil {
 			mt.log.Error(err, "Error occurs during cloning project")
 
 			return err
 		}
-	default:
-		return errors.New(unsupportedConnectionTypeErrorMessage)
-	}
-
-	mt.log.Info(
-		"The model source code was downloaded",
-		"dir", mt.trainerConfig.OutputDir,
-	)
-
-	// Saves some data before starting a training
-	if err := mt.trainClient.SaveModelTrainingResult(
-		k8sTraining.ModelTraining.ID,
-		&odahuflowv1alpha1.TrainingResult{
-			CommitID: commitID,
-		},
-	); err != nil {
-		mt.log.Error(err, "Cannot save the commit id")
 
-		return err
-	}
+		// Saves some data before starting a training
+		if err := mt.trainClient.SaveModelTrainingResult(
+			k8sTraining.ModelTraining.ID,
+			&odahuflowv1alpha1.TrainingResult{
+				CommitID: commitID,
+			},
+		); err != nil {
+			mt.log.Error(err, "Cannot save the commit id")
 
-	mt.log.Info("The commit ID was saved", "commit_id", commitID)
+			return err
+		}
 
-	workDir := mt.trainerConfig.OutputDir
-	if len(workDir) != 0 {
-		mt.log.Info("Change current working dir", "new worker dir", workDir)
-
-		if err := os.Chdir(workDir); err != nil {
-			mt.log.Error(err, "Changing current working dir failed",
-				"new worker dir", workDir,
-			)
+		mt.log.Info("The commit ID was saved", "commit_id", commitID)
+	case connection.S3Type, connection.GcsType, connection.AzureBlobType:
```

Comment:
Let's create a variable `ObjectStorageConnectionTypes` in `packages/operator/pkg/apis/connection/connection.go` of type `map[ConnetionType]struct{}` (analog of set type) and use it instead of enumerating types manually. 
The same variable may be used here: https://github.com/odahu/odahu-flow/pull/542/files#diff-7a5e2da296dad26b0dd21233b73cc4f346fd6284265aa58bb91287418eab983cR220

Reply:
Yeah, i tried it, but it seems like Go doesn't support unpacking slices or any other collections as multiple values for case